src/layout/umap.c

```diff
@@ -666,6 +669,10 @@
         /* Random other nodes are repelled from one (the first) vertex */
         IGRAPH_CHECK(igraph_random_sample(&negative_vertices, 0, no_of_nodes - 2, n_random_vertices));
         for (igraph_integer_t j = 0; j < n_random_vertices; j++) {
+            loop_count++;
+            if (loop_count % interrupt_period == 0) {
```

Comment:
Just my two cents: In terms of speed it typically is faster to do something like `loop_count > interrupt_period` and reset `loop_count = 0` inside the `if`. The modulo is relatively expensive.